presto-main/src/main/java/com/facebook/presto/operator/aggregation/histogram/SingleTypedHistogram.java

```diff
@@ -0,0 +1,343 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.histogram;
+
+import com.facebook.presto.array.IntBigArray;
+import com.facebook.presto.array.LongBigArray;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.type.TypeUtils;
+import com.google.common.base.Suppliers;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.function.Supplier;
+
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INSUFFICIENT_RESOURCES;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static it.unimi.dsi.fastutil.HashCommon.arraySize;
+import static it.unimi.dsi.fastutil.HashCommon.murmurHash3;
+import static java.util.Objects.requireNonNull;
+
+public class SingleTypedHistogram
```

Comment:
idk, but it's a risk i'd rather not take now. i'd prefer a follow-up to measure perf changes 

less code == better when it's as good or equivalent

Reply:
arguably I could remove all the "hash stats" stuff

i used this partly to develop the probing functions when i saw pathological probing

Reply:
never answered, to-do? in the group-by JMC case, I saw single consistently a little faster, but statistically even

leaving as-is for now, but it would be good to have one implementation.

if we wanted to get clever, we could use a state-pattern to toggle single-group vs multi and not track per-group data structures and wrap ValueStore in an interface that would go directly to a BlockBuilder

trade fewer classes for a little more complexity; but done right, it would be easy to read and understand

(a couple classes like ValueStore, not ones created per rwo)